assets/js/plugin-check-admin.js

```diff
@@ -25,17 +27,95 @@
 		);
 	} );
 
+	/**
+	 * Setup the runtime environment if needed.
+	 *
+	 * @since n.e.x.t
+	 */
+	function setUpEnvironment( data ) {
+		const pluginCheckData = new FormData();
+		pluginCheckData.append( 'nonce', pluginCheck.nonce );
+		pluginCheckData.append( 'plugin', data.plugin );
+		pluginCheckData.append( 'action', 'plugin_check_set_up_environment' );
+
+		for (var i = 0; i < data.checks.length; i++) {
+			pluginCheckData.append( 'checks[]', data.checks[ i ] );
+		}
+
+		return fetch(
+			ajaxurl,
+			{
+				method: 'POST',
+				credentials: 'same-origin',
+				body: pluginCheckData
+			}
+		)
+		.then(
+			( response ) => {
+				return response.json();
+			}
+		)
+		.then( handleDataErrors )
+		.then(
+			( data ) => {
+				if ( ! data.data || ! data.data.message ) {
+					throw new Error( 'Response contains no data.' );
+				}
+
+				console.log( data.data.message );
+
+				return data.data;
+			}
+		);
```

Comment:
This code is repeated. Make it into a function or re-usage somehow.

Reply:
@spacedmonkey this is somewhat related to comments in @felixarntz earlier review (https://github.com/10up/plugin-check/pull/101#pullrequestreview-1349321441). There is additional work to be done as part of the remaining issues for milestone 1 so this will change.

Reply:
Created issue. https://github.com/10up/plugin-check/issues/113